File: qqbot-admin-api/src/main/java/com/mieai/qqbot/admin/events/PluginDeliveryPageResponse.java
package com.mieai.qqbot.admin.events;

import java.time.Instant;
import java.util.List;
import java.util.Objects;

public record PluginDeliveryPageResponse(
        List<PluginDeliverySummaryResponse> items,
        String nextCursor,
        boolean hasMore,
        Instant observedAt,
        PluginDeliveryQueueStatsResponse stats) {

    public PluginDeliveryPageResponse {
        items = List.copyOf(Objects.requireNonNull(items, "items must not be null"));
        Objects.requireNonNull(observedAt, "observedAt must not be null");
        Objects.requireNonNull(stats, "stats must not be null");
    }
}

File: qqbot-module-operations/src/main/kotlin/com/mieai/qqbot/admin/events/PluginDeliveryPageResponse.kt
package com.mieai.qqbot.admin.events

import java.time.Instant

data class PluginDeliveryPageResponse private constructor(
    val items: List<PluginDeliverySummaryResponse>,
    val nextCursor: String?,
    val hasMore: Boolean,
    val observedAt: Instant,
    val stats: PluginDeliveryQueueStatsResponse,
    private val normalized: Boolean,
) {
    constructor(
        items: List<PluginDeliverySummaryResponse>,
        nextCursor: String?,
        hasMore: Boolean,
        observedAt: Instant,
        stats: PluginDeliveryQueueStatsResponse,
    ) : this(
        items.toList(),
        nextCursor,
        hasMore,
        observedAt,
        stats,
        true,
    )

}
